src/lib/db.ts

```diff
@@ -546,13 +563,22 @@ export class DB {
     if (this.workspaceRef) {
       this.workspaceRef.off("child_added", this.handleWorkspaceChildAdded);
     }
+
+    Object.keys(this.workspaceModelDisposers).forEach((sectionId) => {
+      this.workspaceModelDisposers[sectionId]();
+    });
   }
 
   private handleWorkspaceChildAdded = (snapshot: firebase.database.DataSnapshot) => {
     const {user, workspaces, ui} = this.stores;
     const sectionDocument: DBOfferingUserSectionDocument|null = snapshot.val();
     if (sectionDocument && !workspaces.getWorkspaceBySectionId(sectionDocument.self.sectionId)) {
       this.createWorkspaceFromSectionDocument(user.id, sectionDocument)
+        .then((workspace) => {
+          this.updateGroupUserSectionDocumentListeners(sectionDocument.self.sectionId);
+          this.monitorWorkspaceVisibility(workspace);
```

Comment:
This series of listener updates is performed after a workspace is opened for the first time or re-loaded; I'm considering moving them into `createWorkspaceFromSectionDocument`